src/lib/utils/ct_utils.h

```diff
@@ -64,15 +67,89 @@ constexpr inline void unpoison(const T* p, size_t n) {
    BOTAN_UNUSED(p, n);
 }
 
+//
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+//
+// Convenience overloads for poisoning/unpoisoning various types and ranges
+//
+
+template <std::integral T>
+constexpr void poison(T& p) {
+   poison(&p, sizeof(T));
```

Comment:
The argument to `poison` is the number of `T` not the bytes

Reply:
Good catch!